samples/invert_tree_charity.py

```diff
@@ -0,0 +1,97 @@
+class Node:
+    """
+    Class Node
+    """
+    def __init__(self, value):
+        self.left = None
+        self.data = value
+        self.right = None
+
+class Tree:
+    """
+    Class tree will provide a tree as well as utility functions.
+    """
+
+    def __init__(self, arr):
+        # create the root
+        self.root = None
+        if (not (type(arr)== list and len(arr) > 0)):
+            return
+        
+        for  i in range (len(arr)):
+            self.root = self.insert(self.root, arr[i])
+        
+
+    def createNode(self, data):
+        """
+        Utility function to create a node.
+        """
+        return Node(data)
+
+    def insert(self, node , data):
+        """
+        Insert function will insert a node into tree.
+        Duplicate keys are not allowed.
+        """
+        #if tree is empty , return a root node
+        if node is None:
+            return self.createNode(data)
+        # if data is smaller than parent , insert it into left side
+        if data < node.data:
+            node.left = self.insert(node.left, data)
+        elif data > node.data:
+            node.right = self.insert(node.right, data)
+
+        return node
+
+
+    def traverse(self, root):
```

Comment:
nit: is there need to supply `root` when it's already in `self.root`?

Reply:
I've created a helper function to help with this